banderwagon/element.go

```diff
@@ -372,6 +396,10 @@ func (p *Element) IsOnCurve() bool {
 
 // IsIdentity returns true if p is the identity element.
 func (p *Element) Normalise() {
+	if p.inner.Z.IsZero() {
+		return
+	}
```

Comment:
IMHO, this was a bug.

If `Z=0`, we can't do the `point_aff.FromProj(&p.inner)`. That code [is here](https://github.com/Consensys/gnark-crypto/blob/master/ecc/bls12-381/bandersnatch/point.go#L239-L244). It doesn't have well-defined behavior for `Z=0`.

In any case, "in real life" (AFAIK) this `Z=0` case must not happen.

Reply:
I think this only happens if its not a banderwagon element, which may imply that the user is abusing the API, so shouldn't we panic here?

Reply:
Yes, if this happens someone is doing something that shouldn't regarding API usage.

Regarding panicking, maybe, but I'm not a huge fan. I'll analyze a bit more if returning an `error` here isn't that annoying and at least let's try to allow the client to shut down gracefully instead of panicking and causing some harm in such a violent stop.

I'll check out and comment again here on what I ended up doing. :)

Reply:
Done, see new review!